packages/util/src/bytes.ts

```diff
@@ -133,123 +170,110 @@ export const unpadHexString = function (a: string): string {
- * Inputs supported: `Buffer`, `String` (hex-prefixed), `Number`, null/undefined, `BigInt` and other objects
- * with a `toArray()` or `toBuffer()` method.
+ * Attempts to turn a value into a `Uint8Array`.
+ * Inputs supported: `Buffer`, `Uint8Array`, `String` (hex-prefixed), `Number`, null/undefined, `BigInt` and other objects
+ * with a `toArray()` or `toBytes()` method.
  * @param v the value
  */
-export const toBuffer = function (v: ToBufferInputTypes): Buffer {
+
+export const toBytes = function (v: ToBytesInputTypes): Uint8Array {
   if (v === null || v === undefined) {
-    return Buffer.allocUnsafe(0)
+    return new Uint8Array()
   }
 
   if (Buffer.isBuffer(v)) {
-    return Buffer.from(v)
+    return Uint8Array.from(v)
   }
 
   if (Array.isArray(v) || v instanceof Uint8Array) {
-    return Buffer.from(v as Uint8Array)
+    return Uint8Array.from(v)
   }
 
   if (typeof v === 'string') {
     if (!isHexString(v)) {
       throw new Error(
-        `Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: ${v}`
+        `Cannot convert string to Uint8Array. toBytes only supports 0x-prefixed hex strings and this string was given: ${v}`
       )
     }
-    return Buffer.from(padToEven(stripHexPrefix(v)), 'hex')
+    return hexToBytes(padToEven(v.slice(2)))
   }
 
   if (typeof v === 'number') {
-    return intToBuffer(v)
+    return intToBytes(v)
   }
 
   if (typeof v === 'bigint') {
     if (v < BigInt(0)) {
-      throw new Error(`Cannot convert negative bigint to buffer. Given: ${v}`)
+      throw new Error(`Cannot convert negative bigint to Uint8Array. Given: ${v}`)
     }
     let n = v.toString(16)
     if (n.length % 2) n = '0' + n
-    return Buffer.from(n, 'hex')
+    return hexToBytes(n)
   }
 
-  if (v.toArray) {
-    // converts a BN to a Buffer
-    return Buffer.from(v.toArray())
-  }
-
-  if (v.toBuffer) {
-    return Buffer.from(v.toBuffer())
+  if (v.toBytes !== undefined) {
+    // converts a `TransformableToBytes` object to a Uint8Array
+    return v.toBytes()
   }
 
   throw new Error('invalid type')
 }
 
 /**
- * Converts a `Buffer` into a `0x`-prefixed hex `String`.
- * @param buf `Buffer` object to convert
+ * Converts a {@link Uint8Array} to a {@link bigint}
  */
-export const bufferToHex = function (buf: Buffer): string {
-  buf = toBuffer(buf)
-  return '0x' + buf.toString('hex')
-}
-
-/**
- * Converts a {@link Buffer} to a {@link bigint}
- */
-export function bufferToBigInt(buf: Buffer) {
-  const hex = bufferToHex(buf)
+export function bytesToBigInt(bytes: Uint8Array) {
+  const hex = bytesToPrefixedHexString(bytes)
   if (hex === '0x') {
     return BigInt(0)
   }
   return BigInt(hex)
 }
 
 /**
- * Converts a {@link bigint} to a {@link Buffer}
+ * Converts a {@link bigint} to a {@link Uint8Array}
  */
-export function bigIntToBuffer(num: bigint) {
-  return toBuffer('0x' + num.toString(16))
+export const bigIntToBytes = (num: bigint) => {
```

Comment:
In this file we have a mix of:

- Standard functions `export function(...)`
- Anonymous functions assigned to variables `export const bytesToInt = function(...)`
- Arrow functions assigned to variables `export const bigIntToBytes = (...) =>`

It doesn't look like we're making use of any idiosyncracies (e.g. this binding), so I don't see a reason for the inconsistency. I'd personally go with `export function bigIntToBytes(...)` but have no strong opinion on that.

Reply:
Probably just accretions from various devs over the years.  My default preference is to always do `export const functionname = () =>...` but that's a weakly held opinion.  The various `this` semantics that accompany arrow functions vs `export function...` are a little fuzzy to me these days so happy to go with whatever.  That all said, I think our general implicit style is to use `export const functionName = () => ...`, at least as long as I've been on the team.